`TestNSMGR_RemoteUsecase_BusyEndpoints` has goroutine leaks sometimes

https://github.com/networkservicemesh/sdk/pull/612/checks?check_run_id=1508552180
```
--- FAIL: TestNSMGR_RemoteUsecase_BusyEndpoints (1.33s)
Error:     leaks.go:78: found unexpected goroutines:
        [Goroutine 638 in state chan send, with github.com/networkservicemesh/api/pkg/api/registry.ReadNetworkServiceChannel.func1 on top of the stack:
        goroutine 638 [chan send]:
        github.com/networkservicemesh/api/pkg/api/registry.ReadNetworkServiceChannel.func1(0x11b1760, 0xc0003ff440, 0xc00067e5a0)
        	/home/runner/go/pkg/mod/github.com/networkservicemesh/api@v0.0.0-20201204203731-4294f67deaa4/pkg/api/registry/helpers.go:54 +0x7a
        created by github.com/networkservicemesh/api/pkg/api/registry.ReadNetworkServiceChannel
        	/home/runner/go/pkg/mod/github.com/networkservicemesh/api@v0.0.0-20201204203731-4294f67deaa4/pkg/api/registry/helpers.go:52 +0x85
        ]
```

Seems like it is not an issue with test, but some issue with discover/watch:
https://github.com/networkservicemesh/sdk/pull/671/checks?check_run_id=1767112466
```
--- FAIL: TestNSMGR_PassThroughLocal (1.67s)
Error:     leaks.go:78: found unexpected goroutines:
        [Goroutine 9393 in state chan send, with github.com/networkservicemesh/sdk/pkg/registry/common/memory.(*networkServiceEndpointRegistryServer).sendEvent.func1 on top of the stack:
        goroutine 9393 [chan send]:
        github.com/networkservicemesh/sdk/pkg/registry/common/memory.(*networkServiceEndpointRegistryServer).sendEvent.func1()
        	D:/a/sdk/sdk/pkg/registry/common/memory/nse_server.go:129 +0x10f
        github.com/edwarnicke/serialize.(*Executor).process(0xc002754208, 0xc000bd6c40)
        	C:/Users/runneradmin/go/pkg/mod/github.com/edwarnicke/serialize@v1.0.7/serialize.go:86 +0x1d9
        created by github.com/edwarnicke/serialize.(*Executor).AsyncExec
        	C:/Users/runneradmin/go/pkg/mod/github.com/edwarnicke/serialize@v1.0.7/serialize.go:56 +0x2e5
        ]
```
https://github.com/networkservicemesh/sdk/pull/671/checks?check_run_id=1763674634
```
--- FAIL: TestNSMGR_SelectsRestartingEndpoint (2.57s)
Error:     leaks.go:78: found unexpected goroutines:
        [Goroutine 2859 in state chan send, with github.com/networkservicemesh/api/pkg/api/registry.ReadNetworkServiceEndpointChannel.func1 on top of the stack:
        goroutine 2859 [chan send]:
        github.com/networkservicemesh/api/pkg/api/registry.ReadNetworkServiceEndpointChannel.func1(0x1dc44a0, 0xc000098d40, 0xc00058e600)
        	/Users/runner/go/pkg/mod/github.com/networkservicemesh/api@v0.0.0-20210112152104-45029fb10e27/pkg/api/registry/helpers.go:66 +0x7a
        created by github.com/networkservicemesh/api/pkg/api/registry.ReadNetworkServiceEndpointChannel
        	/Users/runner/go/pkg/mod/github.com/networkservicemesh/api@v0.0.0-20210112152104-45029fb10e27/pkg/api/registry/helpers.go:64 +0x85
        ]
```

--- a/pkg/networkservice/chains/nsmgr/server_test.go
+++ b/pkg/networkservice/chains/nsmgr/server_test.go
@@ -149,8 +149,6 @@ func TestNSMGR_SelectsRestartingEndpoint(t *testing.T) {
 }
 
 func TestNSMGR_RemoteUsecase_BusyEndpoints(t *testing.T) {
-	t.Skip("https://github.com/networkservicemesh/sdk/issues/619")
-
 	defer goleak.VerifyNone(t, goleak.IgnoreCurrent())
 	ctx, cancel := context.WithTimeout(context.Background(), time.Minute*5)
 	defer cancel()

--- a/pkg/networkservice/common/discover/server.go
+++ b/pkg/networkservice/common/discover/server.go
@@ -102,70 +102,62 @@ func (d *discoverCandidatesServer) discoverNetworkServiceEndpoint(ctx context.Co
 			Name: nseName,
 		},
 	}
+
 	nseStream, err := d.nseClient.Find(ctx, query)
 	if err != nil {
 		return nil, errors.WithStack(err)
 	}
 	nseList := registry.ReadNetworkServiceEndpointList(nseStream)
+
 	if len(nseList) != 0 {
 		return nseList[0], nil
 	}
+
 	query.Watch = true
+
 	nseStream, err = d.nseClient.Find(ctx, query)
 	if err != nil {
 		return nil, errors.WithStack(err)
 	}
-	nseCh := registry.ReadNetworkServiceEndpointChannel(nseStream)
-	select {
-	case <-ctx.Done():
-		return nil, errors.Wrapf(ctx.Err(), "nse: %+v is not found", query.NetworkServiceEndpoint)
-	case nse, ok := <-nseCh:
-		if ok {
-			return nse, nil
-		}
-		return nil, errors.New("nse stream is closed")
-	}
+	return nseStream.Recv()
 }
+
 func (d *discoverCandidatesServer) discoverNetworkServiceEndpoints(ctx context.Context, ns *registry.NetworkService, labels map[string]string) ([]*registry.NetworkServiceEndpoint, error) {
 	query := &registry.NetworkServiceEndpointQuery{
 		NetworkServiceEndpoint: &registry.NetworkServiceEndpoint{
 			NetworkServiceNames: []string{ns.Name},
 		},
 	}
+
 	nseStream, err := d.nseClient.Find(ctx, query)
 	if err != nil {
 		return nil, errors.WithStack(err)
 	}
 	nseList := registry.ReadNetworkServiceEndpointList(nseStream)
 
-	nseList = matchEndpoint(labels, ns, nseList...)
-
-	if len(nseList) != 0 {
-		return nseList, nil
+	result := matchEndpoint(labels, ns, nseList...)
+	if len(result) != 0 {
+		return result, nil
 	}
 
 	query.Watch = true
 
 	ctx, cancelFind := context.WithCancel(ctx)
 	defer cancelFind()
+
 	nseStream, err = d.nseClient.Find(ctx, query)
 	if err != nil {
 		return nil, errors.WithStack(err)
 	}
-	nseCh := registry.ReadNetworkServiceEndpointChannel(nseStream)
 	for {
-		select {
-		case <-ctx.Done():
-			return nil, errors.Wrapf(ctx.Err(), "nse: %+v is not found", query.NetworkServiceEndpoint)
-		case nse, ok := <-nseCh:
-			if ok {
-				result := matchEndpoint(labels, ns, nse)
-				if len(result) != 0 {
-					return result, nil
-				}
-			} else {
-				return nil, errors.New("nse stream is closed")
-			}
+		var nse *registry.NetworkServiceEndpoint
+		if nse, err = nseStream.Recv(); err != nil {
+			return nil, err
+		}
+
+		result = matchEndpoint(labels, ns, nse)
+		if len(result) != 0 {
+			return result, nil
 		}
 	}
 }
@@ -177,28 +169,25 @@ func (d *discoverCandidatesServer) discoverNetworkService(ctx context.Context, n
 			Payload: payload,
 		},
 	}
+
 	nsStream, err := d.nsClient.Find(ctx, query)
 	if err != nil {
 		return nil, errors.WithStack(err)
 	}
 	nsList := registry.ReadNetworkServiceList(nsStream)
+
 	if len(nsList) != 0 {
 		return nsList[0], nil
 	}
+
 	ctx, cancelFind := context.WithCancel(ctx)
 	defer cancelFind()
+
 	query.Watch = true
+
 	nsStream, err = d.nsClient.Find(ctx, query)
 	if err != nil {
 		return nil, errors.WithStack(err)
 	}
-	select {
-	case <-ctx.Done():
-		return nil, errors.Wrapf(ctx.Err(), "ns:\"%v\" with payload:\"%v\" is not found", name, payload)
-	case ns, ok := <-registry.ReadNetworkServiceChannel(nsStream):
-		if ok {
-			return ns, nil
-		}
-		return nil, errors.New("ns stream is closed")
-	}
+	return nsStream.Recv()
 }

